include/qfm/asset/asset_type.hpp

```diff
@@ -5,27 +5,9 @@
 namespace qfm {
 namespace asset {
 
-class AssetType {
- public:
-  explicit AssetType(const std::string& type) noexcept;
-  ~AssetType() = default;
+enum AssetType { CURRENCY, BOND, STOCK, CALL_OPTION, PUT_OPTION, FUTURE };
```

Comment:
Best practice states that this should be `enum class` instead.

See [Enum.3: Prefer class enums over “plain” enums](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#enum3-prefer-class-enums-over-plain-enums) from C++ Core Guidelines.